### PR TITLE
Polish query accepting

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
We shouldn't send the same request to Copilot if the query hasn't changed. So if the query is the same, we short circut.

Fixes https://github.com/microsoft/vscode-internalbacklog/issues/4286

Also, when we open in chat, we should use the last accepted query, not what's in the input box.

Fixes https://github.com/microsoft/vscode-internalbacklog/issues/4280

### CODE REVIEW

File: src/vs/workbench/contrib/chat/browser/actions/chatQuickInputActions.ts [hunk 1]
```diff
@@ -93,8 +94,9 @@ class AskQuickQuestionAction extends Action2 {
 		});
 		disposableStore.add(openInChat);
 		disposableStore.add(openInChat.onChange(async () => {
-			await this._currentSession?.openInChat(this._input!.value);
+			await this._currentSession?.openInChat(this._lastAcceptedQuery ?? this._input!.value);
```
Comment (Tyriar) [inline src/vs/workbench/contrib/chat/browser/actions/chatQuickInputActions.ts:97]: You could avoid the ! if you set a `input` variable to the outer function scope

File: src/vs/workbench/contrib/chat/browser/actions/chatQuickInputActions.ts [hunk 2]
```diff
@@ -249,10 +253,13 @@ class InteractiveQuickPickSession extends Disposable {
 
 	async accept(query: string) {
 		await this._model.waitForInitialization();
+		const requests = this._model.getRequests();
+		const lastRequest = requests[requests.length - 1];
+		if (lastRequest?.message && lastRequest?.message === query) {
```
Comment (Tyriar) [inline src/vs/workbench/contrib/chat/browser/actions/chatQuickInputActions.ts:258]: The `lastRequest?.message` condition is redundant

**Review by Tyriar — APPROVED** [review]
Disabled auto merge in case you want the optional suggestions here

File: src/vs/workbench/contrib/chat/browser/actions/chatQuickInputActions.ts (hunk 1 above)
Comment (TylerLeonhardt) [reply src/vs/workbench/contrib/chat/browser/actions/chatQuickInputActions.ts:97]: will add in debt week

File: src/vs/workbench/contrib/chat/browser/actions/chatQuickInputActions.ts (hunk 2 above)
Comment (TylerLeonhardt) [reply src/vs/workbench/contrib/chat/browser/actions/chatQuickInputActions.ts:258]: will add in debt week

**Pull request merged**